packages/core/src/electron-browser/menu/electron-menu-contribution.ts

```diff
@@ -207,7 +207,11 @@ export class ElectronMenuContribution implements FrontendApplicationContribution
         registry.registerCommand(ElectronCommands.TOGGLE_FULL_SCREEN, {
             isEnabled: () => currentWindow.isFullScreenable(),
             isVisible: () => currentWindow.isFullScreenable(),
-            execute: () => currentWindow.setFullScreen(!currentWindow.isFullScreen())
+            execute: () => {
+                currentWindow.setFullScreen(!currentWindow.isFullScreen());
+                currentWindow.menuBarVisible = true; // Ensure menu remains visible on Linux
```

Comment:
@westbury as I mentioned in the issue itself, the behavior is controllable in vscode through the `window.menuBarVisibility` preference:

- `window.menuBarVisibility: classic`: will hide the menu-bar on fullscreen
- `window.menuBarVisibility: visible`: will keep the menu-bar visible on fullscreen

You can confirm by testing with vscode.

Reply:
```suggestion
                currentWindow.menuBarVisible = this.preferenceService.get('window.menuBarVisibility', 'classic') === 'visible';
```